src/rust/engine/src/externs/options.rs

```diff
@@ -146,6 +149,50 @@ impl PyOptionId {
     }
 }
 
+#[pyclass]
+struct PySplitArgs(SplitArgs);
+
+#[pymethods]
+impl PySplitArgs {
+    fn goals(&self) -> &Vec<String> {
+        &self.0.goals
+    }
+
+    fn unknown_goals(&self) -> &Vec<String> {
+        &self.0.unknown_goals
+    }
+
+    fn specs(&self) -> &Vec<String> {
+        &self.0.specs
+    }
+
+    fn passthru(&self) -> &Vec<String> {
+        &self.0.passthru
+    }
+}
+
+#[pyclass]
+struct PyArgSplitter(ArgSplitter);
+
+#[pymethods]
+impl PyArgSplitter {
+    #[new]
+    fn __new__(build_root: &str, known_goal_names: Vec<String>) -> PyResult<Self> {
+        Ok(Self(ArgSplitter::new(
+            &PathBuf::from(build_root),
```

Comment:
Maybe just construct `&Path` directly with [`Path::new(build_root)`](https://doc.rust-lang.org/std/path/struct.Path.html#method.new)?

Reply:
Also, this can just return `Self` since the method always returns `Ok`.

Reply:
Good catch, in an earlier iteration this was able to return an Error, but not now.